src/traces/scatter/plot.js

```diff
@@ -483,6 +485,8 @@ function plotOne(gd, idx, plotinfo, cdscatter, cdscatterAll, element, transition
                 });
             });
 
+        join.order();
```

Comment:
:sos: IMPORTANT :sos: 

This commit breaks `scatter/plot.js` for traces with non-numeric (x,y) positions. 

That is due to [`Drawing.translatePoint`](https://github.com/plotly/plotly.js/blob/cea9455ba353fe62ef374454cc954c4b7945bb0a/src/components/drawing/index.js#L74) which is called on the [enter](https://github.com/plotly/plotly.js/blob/cea9455ba353fe62ef374454cc954c4b7945bb0a/src/traces/scatter/plot.js#L423) and [merge](https://github.com/plotly/plotly.js/blob/cea9455ba353fe62ef374454cc954c4b7945bb0a/src/traces/scatter/plot.js#L433) selections. When points with non-numeric (x,y) are detected, `Drawing.translatePoint` removes them. Unfortunately, this makes the `join` selections out-of-sync with the DOM and leads to _`'TypeError: Cannot read property 'textContent' of null`_ exception thrown by d3.

----

So, I have two questions for the d3-heads looking at this PR:

- Is there a way to re-select the join and call `order()` on that selection? 
- Or should we make our updates more d3-esque by filtering data items with non-numeric position **before** binding them to the selections?